src/vcpkg/platform-expression.cpp

```diff
@@ -139,26 +142,114 @@ namespace vcpkg::PlatformExpression
+                }
+            }
+
+            // platform-expression-and =
+            // | platform-expression-not, { "&", optional-whitespace, platform-expression-not } ;
+            // | platform-expression-not, { "and", optional-whitespace, platform-expression-not } ;
+            //
+            // platform-expression-or =
+            // | platform-expression-not, { "|", optional-whitespace, platform-expression-not } ;
+            //
+            // platform-expression-low-precedence-or =
+            // | platform-expression-not, { ",", optional-whitespace, platform-expression-not } ;
+            ExprKind expr_operator()
+            {
+                auto oper = cur();
+
+                // Support chains of the vcpkg operators (`&`, `|`)  to avoid breaking backwards compatibility
+                switch (oper)
+                {
+                    case '|':
+                    case '&':
+                        do
+                        {
+                            next();
+                        } while (allow_multiple_binary_operators() && cur() == oper);
+                        break;
+                }
+
+                switch (oper)
                 {
                     case '|':
                     {
                         // { "|", optional-whitespace, platform-expression-not }
-                        return expr_binary<'|', '&'>(std::make_unique<ExprImpl>(ExprKind::op_or, std::move(result)));
+                        return ExprKind::op_or;
                     }
                     case '&':
                     {
                         // { "&", optional-whitespace, platform-expression-not }
-                        return expr_binary<'&', '|'>(std::make_unique<ExprImpl>(ExprKind::op_and, std::move(result)));
+                        return ExprKind::op_and;
+                    }
+                    case ',':
+                    {
+                        // { ",", optional-whitespace, platform-expression-not }
+                        // "," is a near-synonym of "|", with the differences that it can be combined with "&"/"and",
+                        // but has lower precedence
+                        next();
+                        return ExprKind::op_low_precedence_or;
+                    }
+                    case 'a':
+                    {
+                        // { "and", optional-whitespace, platform-expression-not }
+                        // "and" is a synonym of "&"
+                        std::string name = match_zero_or_more(is_identifier_char).to_string();
+                        if (!name.empty() && (name == "and"))
+                        {
+                            return ExprKind::op_and;
+                        }
+
+                        // Invalid alphanumeric strings or strings other than "and" are errors.
+                        if (name.empty())
+                        {
+                            add_error("unexpected character in logic expression");
```

Comment:
```suggestion
                            Checks::unreachable(VCPKG_LINE_INFO);
```
Because we matched an 'a' in the case statement, it should be impossible for `name` to be empty (if I'm reading the code correctly).

Reply:
You are correct. At that point in the code, cur() is 'a' and based on the logic in match_zero_or_more(), at least the 'a' must be returned. I will modify the code to remove the unnecessasry name.empty() checks.